hydra-explorer/src/Hydra/Explorer/ExplorerState.hs

```diff
@@ -2,20 +2,24 @@ module Hydra.Explorer.ExplorerState where
 
 import Hydra.Prelude
 
-import Hydra.Tx.HeadId (HeadId (..), HeadSeed)
+-- XXX: This is the only proper dependency onto hydra-node. Should factor this
+-- into a hydra-chain package with smaller dependency footprint.
+import Hydra.Chain (OnChainTx (..))
+
+-- XXX: Need to depend on hydra-tx:testlib for generators?
```

Comment:
@locallycompact You moved the generators into a `testlib` component. What is your opinion here?

Reply:
This could be `deriving anyclass Generic`

At the moment I don't mind anything. All options feel wrong for one reason or another, but given that two of the major dependencies of hydra both use hedgehog (cardano-ledger and cardano-api), it feels like the path of least resistance to just align there, [switch to hedgehog](https://github.com/cardano-scaling/hydra/issues/1758) and use named generators uniformly across all hydra projects in a sublibrary (either :gen or :testlib). Arbitrary classes, locally defined, feel great when everything is naturally generic, but when not it is inconsistent since we then need to rely on named generators.

Reply:
I meant not to ask about Arbitrary vs named generators nor even QuickCheck vs hedgehog.

My question is rather.. why do generators need to be a `:testlib`? Doesn't it feel weird to depend on it from an application package?

Reply:
Sublibraries are just another kind of way that we have of selectively ignoring information. I think generators lend themselves well to being sublibraries since a) they typically have unique dependencies themselves distinct from the rest of the code, b) they are used generally only for testing, and noise otherwise when worrying about logic.

Reply:
I don't think we should engineer it such that we depend on a testlib from an application. We could never need to do that.

Reply:
So your suggestion is to move the `Arbitrary ExplorerState` instance (for which we need this import here) into yet another `hydra-explorer:testlib` component?

I see now why you brought up the hedgehog and moving away from `Arbitrary` classes, because moving things into a `testlib` here would be at odds with not wanting orphan instances (https://github.com/cardano-scaling/hydra/wiki/Coding-Standards#proposal-avoid-orphan-instances)